.github/workflows/codeql.yaml

```diff
@@ -0,0 +1,108 @@
+        #
+        # Use `c-cpp` to analyze code written in C, C++ or both.
+        # Use 'java-kotlin' to analyze code written in Java, Kotlin or both.
+        # Use 'javascript-typescript' to analyze code written in JavaScript,
+        # TypeScript or both.
+        #
+        # To learn more about changing the languages that are analyzed or
+        # customizing the build mode for your analysis, see:
+        # https://docs.github.com/en/code-security/code-scanning/creating-an-advanced-setup-for-code-scanning/customizing-your-advanced-setup-for-code-scanning.
+        #
+        # If you are analyzing a compiled language, you can modify the
+        # 'build-mode' for that language to customize how your codebase is
+        # analyzed, see:
+        # https://docs.github.com/en/code-security/code-scanning/creating-an-advanced-setup-for-code-scanning/codeql-code-scanning-for-compiled-languages
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+
+    # Add any setup steps before running the `github/codeql-action/init` action.
+    # This includes steps like installing compilers or runtimes
+    # (`actions/setup-node` or others). This is typically only required for
+    # manual builds.
+    # - name: Setup runtime (example)
+    #   uses: actions/setup-example@v1
+
+    # Initializes the CodeQL tools for scanning.
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v3
+      with:
+        languages: ${{ matrix.language }}
+        build-mode: ${{ matrix.build-mode }}
+        # If you wish to specify custom queries, you can do so here or in a
+        # config file.
+        # By default, queries listed here will override any specified in a
+        # config file.
+        # Prefix the list here with "+" to use these queries and those in the
+        # config file.
+
+        # For more details on CodeQL's query packs, refer to:
+        # https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/configuring-code-scanning#using-queries-in-ql-packs
+        # queries: security-extended,security-and-quality
+
+    # If the analyze step fails for one of the languages you are analyzing with
+    # "We were unable to automatically build your code", modify the matrix above
+    # to set the build mode to "manual" for that language. Then modify this step
+    # to build your code.
+    # Command-line programs to run using the OS shell.
+    # See https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsrun
+    - if: matrix.build-mode == 'manual'
+      shell: bash
+      run: |
+        cat <<'...'
+        If you are using a "manual" build mode for one or more of the languages
+        you are analyzing, replace this with the commands to build your code.
+
+        For example:
+          make bootstrap
+          make release
+        ...
+        exit 1
```

Comment:
Looks like this can be removed, unless the project doesn't automatically build? And in that case we'd put actual build steps?

Reply:
I read
- https://docs.github.com/en/code-security/code-scanning/creating-an-advanced-setup-for-code-scanning/codeql-code-scanning-for-compiled-languages#about-specifying-build-steps-manually
- and https://docs.github.com/en/code-security/code-scanning/creating-an-advanced-setup-for-code-scanning/codeql-code-scanning-for-compiled-languages#building-go

and honestly don't really get what this does. is this a text area you can configure when you run the action manually?

Reply:
Fixed in 9487260